### PR TITLE
[a11y] [com_fields] icons in modals

### PR DESCRIPTION
**Pull request by brianteeman**
Continuing the work to ensure that screen readers dont try to read the value of a font icon.

This PR could be improved with the use of titles as well if anyone has some suggestions to get the requires strings

### CODE REVIEW

File: administrator/components/com_categories/views/categories/tmpl/modal.php [hunk 1]
```diff
@@ -107,7 +107,7 @@
 						?>
 						<tr class="row<?php echo $i % 2; ?>">
 							<td class="center">
-								<span class="<?php echo $iconStates[$this->escape($item->published)]; ?>"></span>
+							<span class="<?php echo $iconStates[$this->escape($item->published)]; ?>" aria-hidden="true"></span>
```
Comment (Quy) [inline administrator/components/com_categories/views/categories/tmpl/modal.php]: Revert tab deletion.

Comment (brianteeman) [inline administrator/components/com_categories/views/categories/tmpl/modal.php]: good spot thanks - fixed

**Comment by tonypartridge** [comment]
I have tested this item :white_check_mark: successfully on 96e15759d9f8c9902b53f3999f2812e65b923bfd<br /><br />Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15047">issues.joomla.org/tracker/joomla-cms/15047</a>.</sub>

**Comment by Quy** [comment]
Please provide instructions on how to invoke these modals. Thanks.

**Comment by Quy** [comment]
```
PHP Notice:  Undefined property: stdClass::$published in \administrator\components\com_content\views\articles\tmpl\modal.php on line 122

PHP Notice:  Undefined index:  in \administrator\components\com_content\views\articles\tmpl\modal.php on line 122
```

File: administrator/components/com_content/views/articles/tmpl/modal.php [hunk 2]
```diff
@@ -119,7 +119,7 @@
 					?>
 					<tr class="row<?php echo $i % 2; ?>">
 						<td class="center">
-							<span class="<?php echo $iconStates[$this->escape($item->state)]; ?>"></span>
+							<span class="<?php echo $iconStates[$this->escape($item->published)]; ?>" aria-hidden="true"></span>
```
Comment (Quy) [inline administrator/components/com_content/views/articles/tmpl/modal.php]:
```
PHP Notice:  Undefined property: stdClass::$published in \administrator\components\com_content\views\articles\tmpl\modal.php on line 122

PHP Notice:  Undefined index:  in \administrator\components\com_content\views\articles\tmpl\modal.php on line 122
```

**Comment by Quy** [comment]
Here is an idea to get the status strings:
```
				<?php
				$iconStates = array(
					-2 => array('icon-trash', JText::_('JTRASHED')),
					0 => array('icon-unpublish', JText::_('JUNPUBLISHED')),
					1 => array('icon-publish', JText::_('JPUBLISHED')),
					2 => array('icon-archive', JText::_('JARCHIVED')),
				);
				?>
```
```
						<td class="center">
							<span class="<?php echo $iconStates[$this->escape($item->state)][0]; ?>" aria-hidden="true"></span> <span class="element-invisible"><?php echo $iconStates[$this->escape($item->state)][1]; ?></span>
						</td>
```

File: administrator/components/com_associations/views/associations/tmpl/modal.php [hunk 3]
```diff
@@ -106,7 +106,7 @@
 				<tr class="row<?php echo $i % 2; ?>">
 					<?php if (!empty($this->typeSupports['state'])) : ?>
 						<td class="center">
-							<span class="<?php echo $iconStates[$this->escape($item->state)]; ?>"></span>
+							<span class="<?php echo $iconStates[$this->escape($item->published)]; ?>" aria-hidden="true"></span>
```
Comment (Quy) [inline administrator/components/com_associations/views/associations/tmpl/modal.php]: Revert.

File: administrator/components/com_fields/views/fields/tmpl/modal.php [hunk 4]
```diff
@@ -91,7 +91,7 @@
 					?>
 					<tr class="row<?php echo $i % 2; ?>">
 						<td class="center">
-							<span class="<?php echo $iconStates[$this->escape($item->state)]; ?>"></span>
+							<span class="<?php echo $iconStates[$this->escape($item->published)]; ?>" aria-hidden="true"></span>
```
Comment (Quy) [inline administrator/components/com_fields/views/fields/tmpl/modal.php]: Revert `$iconStates[$this->escape($item->published)]`.

File: administrator/components/com_associations/views/associations/tmpl/modal.php (hunk 3 above)
Comment (brianteeman) [inline administrator/components/com_associations/views/associations/tmpl/modal.php]: oops - copy paste error

**Comment by brianteeman** [comment]
typos fixed - thanks
@quy I will leave that suggestion for another PR

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 5b2b30a9e180d7db78834e6964f75d7b8522d119<br /><br />Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15047">issues.joomla.org/tracker/joomla-cms/15047</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
@tonypartridge can you please retest?

**Comment by brianteeman** [comment]
Updated branch - no need for new tests

**Comment by tonypartridge** [comment]
anyone have any idea why AppVayor is stuck on waiting? @wilsonge ?

**Comment by zero-24** [comment]
I have just gived it another try.

**Comment by wilsonge** [comment]
Merged on review

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**